### PR TITLE
Add version info to build.gradle

### PR DESCRIPTION
**Pull request by andrross**
I copied this approach from other plugin repos, for example [common-utils][1].

[1]: https://github.com/opensearch-project/common-utils/blob/7736e080f9b0a2f6f2d9ad3b4272090b281ea8fa/build.gradle#L43-L52

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,6 +41,17 @@ repositories {
   maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
 }
 
+allprojects {
+  group 'org.opensearch.index.codec.customcodecs'
```
Comment (reta) [inline build.gradle:45]: Shouldn't the group be org.opensearch.plugin (https://github.com/opensearch-project/OpenSearch/blob/main/plugins/build.gradle)?

Comment (andrross) [reply build.gradle:45]: I'm honestly not sure. I followed the convention here: https://github.com/opensearch-project/common-utils/blob/main/build.gradle#L44

**Comment by gaiksaya** [comment]
This should help test it out https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#publishing-java-plugins-snapshots

File: build.gradle (hunk 1 above)
Comment (andrross) [reply build.gradle:45]: I think it should be `org.opensearch` per the example here: https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#define-a-version-based-on-the-opensearch-dependency

Comment (gaiksaya) [reply build.gradle]: Right https://github.com/opensearch-project/index-management/blob/main/build.gradle#L192

**Comment by andrross** [comment]
I'm merging this now to unblock initial integration testing. We can revisit the specific values here if they turn out to be incorrect.

**Pull request merged**